tests/core/toolkit/test_destroyable_thread.py

```diff
@@ -38,11 +38,22 @@ def infinite_wait():
             time.sleep(0.1)
 
     thread = DestroyableThread(infinite_wait)
-
     thread.start()
     thread.destroy()
+    assert thread.canceled
     thread.join(60)
+    assert not thread.is_alive()
+
+
+def test_threads_canceled_when_interrupt_fails():
+    def long_sleep():
+        time.sleep(0.2)
 
+    thread = DestroyableThread(long_sleep)
+    thread.start()
+    thread.destroy()
+    assert thread.canceled
```

Comment:
Is this trying to assert that the `canceled` flag is set even when the thread itself is not interrupted?
I wonder if this can be done with a wait on an event so that we don't have to jam a `sleep` in here and have the race conditions of:
- Thread has not started yet when `destroy` is called
- Thread has already finished when `destroy` is called
...and also have to wait the entire `0.2` seconds for the test to finish. That's like a solid 5% of our entire unit test runtime atm

Reply:
Interesting, good call on using events here. I'll experiment a bit

Reply:
So I think this is a good approach to eliminating the possibility that this test is flakey, but I also think we need to keep the `sleep(0.2)` in there since the whole point of this test is to simulate `canceled` being true when the thread has not yet died completely. If it's sitting waiting on an event, it will happily raise the internal exception and die as soon as `destroy` is called.

Reply:
I'll go ahead and remove the `assert thread.is_alive()` as that's the one that could fail as currently written if the races come up